### ISSUE TITLE
Add `<SearchFiltersPanel>` and children into `kolibri-common`

### ISSUE THREAD

**Issue by marcellamaki** [learningequality/kolibri#12519]
Prerequisite: #12609 

**Tasks**
- [ ] Add `SearchFiltersPanel` component into `kolibri-common` 
- [ ] Add all children components that support it
- [ ] Add <SearchResultsGrid> into kolibri-common 
- [ ] Move relevant related components (such as <LearningActivityChip>)
- [ ] Update all file paths in the `kolibri-common` version
- [ ] Update all references in Learn to use the `kolibri-common` version, and ensure everything still builds and works as expected; clean up old files in Learn
- [ ] Document that because the `SearchFiltersPanel` uses `injectSearch()`, that wherever it is used, it requires `useSearch()` in the component ancestry

**Out of scope**
- Refactoring the grid to use the new grid system and `KCard` will be done after this; this task is only about moving the components and ensuring

**Acceptance Criteria**
- `Learn` plugin should be regression tested by the assignee and by the QA team and everything should be working as expected 
- Code comments should be added in the `SearchFiltersPanel` -- for support on how to explain or how much detail is required, chat with Richard or Marcella
